code/ui/WeaponSelection.cs

```diff
@@ -1,37 +1,188 @@
-//using System.Collections.Generic;
+using System.Linq;
+using System.Collections.Generic;
+using Sandbox;
 using Sandbox.UI;
 using Sandbox.UI.Construct;
 
+using TTTReborn.Player;
+using TTTReborn.Weapons;
+
+// TODO Fix animation on dropping a higher slot weapon (instead of deleting and recreating, move and delete WeaponSlots in DOM)
+
 namespace TTTReborn.UI
 {
     public class WeaponSelection : Panel
     {
-        //List<WeaponSlot> weaponSlots = new();
+        private Dictionary<string, WeaponSlot> weaponSlots = new();
+
+        private Weapon oldActiveWeapon;
 
         public WeaponSelection()
         {
             StyleSheet.Load("/ui/WeaponSelection.scss");
+        }
+
+        // TODO: This method could be made to be event based. Whenever a user pickups a weapon, and whenever a user fires.
+        // Consider checking out DM98 or Hidden inventory HUD.
+        public override void Tick()
+        {
+            base.Tick();
+
+            var player = Local.Pawn as TTTPlayer;
+
+            if (player == null)
+            {
+                return;
+            }
+
+            // update weapon slots, check for removed weapons etc.
+            Inventory inventory = player.Inventory as Inventory;
+            int inventoryCount = inventory.Count();
+            List<Weapon> toAdd = new();
+            List<WeaponSlot> tmpSlots = new();
```

Comment:
Hmm could be more descriptive names, trying to come up with myself too.
```csharp
List<Weapon> newWeapons = new();
List<WeaponSlot> ...
```

Reply:
Yea, but that's still a dirty way. Theoretically, we have to fetch which weapon was dropped (per RPCs?), then remove the corresponding WeaponSlot and same if a weapon was added (sort the DOM again, somehow ^^)